packages/@lwc/engine-core/src/framework/decorators/register.ts

```diff
@@ -251,10 +260,12 @@ export function registerDecorators(
             descriptor = getOwnPropertyDescriptor(proto, fieldOrMethodName);
             if (method === 1) {
                 if (process.env.NODE_ENV !== 'production') {
-                    assert.isTrue(
-                        adapter,
-                        `@wire on method "${fieldOrMethodName}": adapter id must be truthy.`
-                    );
+                    if (!adapter) {
+                        // TODO [#3408]: this should throw, not log
+                        logError(
+                            `@wire on method "${fieldOrMethodName}": adapter id must be truthy.`
+                        );
+                    }
```

Comment:
Despite the name, `assert.isTrue()` actually asserts _truthiness_, not `=== true`. So this logic is the same as before.